modules/concurrency/include/hephaestus/concurrency/spinner_manager.h

```diff
@@ -0,0 +1,38 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <future>
+#include <vector>
+
+#include "hephaestus/concurrency/spinner.h"
+
+namespace heph::concurrency {
+
+/// @brief SpinnersManager allows to orchestrate the execution of multiple spinners.
+/// The main feature is `waitAny` which allows to unblock as soon as one of the spinners is done.
+/// This allows to catch errors in spinner execution as soon as possible and stop the others.
+/// NOTE: right now we do not have any concept of error for the spinners: we cannot know if a spinner
+/// terminated with an error or not. If an exception is thrown inside the runner, it will be re-throwed when
+/// we call runner.stop().get(). We leave it to the user to handle it outside of the runner manager.
+/// NOTE: this logic is quite generic and can be extended to any type of object that has `wait()` and `stop()`
+/// methods. To be faitful to the principle of implement only what you need now, we limit the scope to
+/// spinners and consider to expand the scope when an use case arises.
+class SpinnersManager {
+public:
+  explicit SpinnersManager(std::vector<Spinner*> spinners);
+
+  void startAll();
+  void waitAll();
+  void waitAny();
```

Comment:
```suggestion
  /// @brief blocks until all spinners are finished. If a single spinner throws an exception, it will not be propagated as the other spinners are still blocking.
  void waitAll();
  /// @brief wait until any spinner terminates or throws an exception, which allows for immediate exception handling.
  void waitAny();
```
The naming is clear, but given this is somewhat complex async behavior, I would add these two lines. I'm not sure waitAll is useful tbh, as it only is useful if the spinner terminates on its own AND does never throw, which seems a non-existing case. While it is elegant and adds completeness, I feel like removing it.

Reply:
you are right, it could be indeed an early generalization having `waitAll`. 
The current problem with this interface is that from outside (until we call `.get()`) we have no way to know if something was terminated with an exception. In addition, it is not given that just because one of the spinners terminated with an exception you want to stop all of them.
For this reason, you also want to assume there are cases where you want to wait for all the spinner to complete. 
Given the simplicity of the implementation and the fact that it doesn't have impact to the overall design I would keep `waitAll`. WDYT?

Reply:
sounds good